Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -5466,13 +5466,14 @@ std::pair<LayoutUnit, LayoutUnit> RenderBox::computeMinMaxLogicalWidthFromAspect
     ASSERT(style().hasAspectRatio());
     LayoutUnit transferredMinSize = LayoutUnit();
     LayoutUnit transferredMaxSize = LayoutUnit::max();
+    double aspectRatio = boxAspectRatio();
     if (style().logicalMinHeight().isSpecified()) {
         if (LayoutUnit blockMinSize = constrainLogicalHeightByMinMax(LayoutUnit(), std::nullopt); blockMinSize > LayoutUnit())
-            transferredMinSize = inlineSizeFromAspectRatio(horizontalBorderAndPaddingExtent(), verticalBorderAndPaddingExtent(), style().logicalAspectRatio(), style().boxSizingForAspectRatio(), blockMinSize);
+            transferredMinSize = inlineSizeFromAspectRatio(horizontalBorderAndPaddingExtent(), verticalBorderAndPaddingExtent(), aspectRatio, style().boxSizingForAspectRatio(), blockMinSize);
```

Comment:
Yeah so that is actually why I added boxAspectRatio. For most objects, style().logicalAspectRatio() would be fine, but for RenderVideo it depends on whether we have the natural aspect-ratio or not. RenderView overrides it and uses hasVideoMetadata() to distinguish whether or not we have the default object size since the default object size is used before the metadata is received.

Instead of having boxAspectRatio, should we instead move the logic here to determine which aspect ratio to use like in RenderVideo::boxAspectRatio?

Reply:
Hmm, double-checked, the specification and the code. It's fine to use `style().logicalAspectRatio()` here.
First, the value of `style().logicalAspectRatio()` is always `w / h`, it does not consider `aspect-ratio: auto w / h` scenario.
Then, per [1], the value auto && [<ratio>](https://www.w3.org/TR/css-values-4/#ratio-value) only affects the replaced elements. And it looks like `RenderBox::computeMinMaxLogicalWidthFromAspectRatio` does not handle replaced elements at all.
So it seems fine to use `style().logicalAspectRatio()` here. `auto w / h` should be same to `w / h` for non-replaced elements. Do you agree with me here?

As to the replaced element, like img and video:
It is handled in `RenderReplaced::computeReplacedLogicalWidth`, looks like it does not consider the transferred min/max sizes at all. Maybe we need a patch to handle it here?  Maybe in `RenderBox::computeReplacedLogicalWidthRespectingMinMaxWidth`?

As to RenderVideo:
Yes, `hasVideoMetadata()` could distinguish it is using the default size or not. However, I think it might be nature to handle it in `RenderReplaced::computeAspectRatioInformationForRenderBox`, i.e, don't let "the default size" affect `intrinsicRatio`.

[1] https://www.w3.org/TR/css-sizing-4/#preferred-aspect-ratio

Reply:
Yes you are right about the code in RenderBox. I removed the `boxAspectRatio `method so it is calling `style().logicalAspectRatio()` again. 

Since `RenderReplaced::computeReplacedLogicalWidth` is called by `RenderVideo::computeReplacedLogicalWidth`, I added a call to `computeReplacedLogicalWidthRespectingMinMaxWidth` inside of the RenderVideo code which should solve that issue I think.

For RenderVideo, I added a comment to the additional one you had here: https://github.com/WebKit/WebKit/pull/4509#discussion_r983852491